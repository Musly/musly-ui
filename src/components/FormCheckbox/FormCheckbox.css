.checkbox {
  display: flex;
  align-items: center;
  cursor: pointer;

  &:hover {
    .checkbox__label {
      @apply text-gray-600;
    }

    .checkbox__input:not(:disabled) {
      @apply border-blue-300;
    }
  }

  &__input {
    cursor: pointer;
    width: 1.1428rem;
    height: 1.1428rem;
    transition-property: border-color;
    transition-duration: 100ms;

    @apply form-checkbox text-blue-500 rounded border border-gray-400 ease-sharp focus:border-blue-500;

    &:disabled {
      cursor: default;

      @apply border-gray-200;
    }
  }

  &__label {
    font-weight: 500;
    transition-property: color;
    transition-duration: 200ms;

    @apply text-gray-500 ml-2 ease-sharp;
  }
}
