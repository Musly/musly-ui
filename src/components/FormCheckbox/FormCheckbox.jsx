/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import PropTypes from 'prop-types';
import './FormCheckbox.css';

export function FormCheckbox (props) {
  const {
    checked, label, onChange, ...rest
  } = props;

  return (
    <label className="checkbox group">
      <input
        type="checkbox"
        {...rest}
        checked={checked}
        onChange={onChange}
        aria-label={label}
        className="checkbox__input"
      />
      {label && (
        <span className="checkbox__label">
          {label}
        </span>
      )}
    </label>
  );
}

FormCheckbox.propTypes = {
  label: PropTypes.string,
  onChange: PropTypes.func,
  checked: PropTypes.bool,
};

FormCheckbox.defaultProps = {
  label: null,
  onChange: null,
  checked: false,
};
