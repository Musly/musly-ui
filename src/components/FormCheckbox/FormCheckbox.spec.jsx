import React from 'react';
import { shallow } from 'enzyme';
import { FormCheckbox } from './FormCheckbox';

describe('<FormCheckbox />', () => {
  it('should render without label', () => {
    const onChange = jest.fn();
    const result = shallow(<FormCheckbox onChange={onChange} value="some" />);
    expect(result.find('.checkbox__input')).toExist();
    expect(result.find('.checkbox__label')).not.toExist();
    result.find('.checkbox__input').simulate('change');
    expect(onChange).toHaveBeenCalled();
    expect(result).toMatchSnapshot();
  });

  it('should rener with label', () => {
    const result = shallow(<FormCheckbox label="test label" value="some" />);
    expect(result.find('.checkbox__input')).toExist();
    expect(result.find('.checkbox__label')).toExist();
    expect(result.find('.checkbox__label').text()).toEqual('test label');
    expect(result).toMatchSnapshot();
  });
});
