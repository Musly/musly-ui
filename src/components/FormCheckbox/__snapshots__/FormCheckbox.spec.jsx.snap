// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormCheckbox /> should render without label 1`] = `
<label
  className="checkbox group"
>
  <input
    aria-label={null}
    checked={false}
    className="checkbox__input"
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    type="checkbox"
    value="some"
  />
</label>
`;

exports[`<FormCheckbox /> should rener with label 1`] = `
<label
  className="checkbox group"
>
  <input
    aria-label="test label"
    checked={false}
    className="checkbox__input"
    onChange={null}
    type="checkbox"
    value="some"
  />
  <span
    className="checkbox__label"
  >
    test label
  </span>
</label>
`;
